packages/contracts-bedrock/scripts/Deploy.s.sol

```diff
@@ -360,7 +366,17 @@ contract Deploy is Deployer {
         initializeOptimismMintableERC20Factory();
         initializeL1CrossDomainMessenger();
         initializeL2OutputOracle();
-        initializeOptimismPortal();
+
+        // Selectively initialize either the original OptimismPortal or the
+        // new OptimismPortal2. Since this will upgrade the proxy, we cannot
+        // initialize both. FPAC warning can be removed once we're done with
+        // the old OptimismPortal contract.
+        if (cfg.useOptimismPortal2()) {
```

Comment:
smol nit: having the config based on `UseOptimismPortal2` isn't a clear way to communicate that its a switch to use fault proofs or not. A better way to config would be `UseFaultProofs` as the config. This prevents leakage of implementation details in the config and ensures that we config with feature flags cleanly

no real need to massively refactor but just saying for next time